crates/rate-limit/src/lib.rs

```diff
@@ -0,0 +1,3 @@
+pub mod rate_limiter;
+
+pub use rate_limiter::{Error, RateLimiter, Strategy};
```

Comment:
For such a tiny rate I think it's probably better to just throw everything in `lib.rs`.
This just seems a little nicer to me to more easily navigate the code base.